apps/teams-test-app/src/components/privateApis/ChatAPIs.tsx

```diff
@@ -35,10 +35,33 @@ const OpenChat = (): React.ReactElement =>
     },
   });
 
-const OpenGroupChat = (): React.ReactElement =>
+const OpenChat = (): React.ReactElement =>
+  ApiWithTextInput<OpenSingleChatRequest>({
+    name: 'openChat2',
+    title: 'Open Chat',
+    onClick: {
+      validateInput: (input) => {
+        if (!input.user || !input.message) {
+          throw new Error('Both user and message are required on the input');
+        }
+      },
+      submit: async (input) => {
+        return await chat
```

Comment:
Using `async`/`await` AND `.then().catch()` seems wrong. I think you could just remove the `async`/`await` part here, or if you want to continue using it, it should look more like this:

```
try {
    await chat.openChat(input);
    return 'openChat was called';
} catch () {
    return 'openChat - error';
}
```

Reply:
Same for the other new elements you're adding.

Reply:
I tried doing this but got some errors saying it's not returning in all cases, could you help navigate the error? I tried adding another return after the try catch but for some reason it would return the success case and then return the last return as well. (How does that work anyways?? Shouldn't it exit the function as soon as it returns from the try catch? Weird)

Reply:
Oh wait, my bad- it was another case where I would see both the returns happen. In this case, I don't see the "you're not returning error happening", let me try it all the way through.

Reply:
I now can remember that we wanted to stay away from using both async/await and .then/.catch at the same time, but I am not sure that if `chat.openChat(input)` rejects, that it'll go to the catch block. I think it's possible that it expects the reject to be caught only through `.catch(() => {})` right on the promise, but I'm having trouble googling the answer. I can see that if the API is not supported by the hub sdk that it would go into the catch block however.

Reply:
Chatted offline, it did indeed turn out to be a trivial thing to test for, thanks!